File: rust/src/models/metric_list_response.rs
// this file is @generated
use serde::{Deserialize, Serialize};

use super::{metric_summary::MetricSummary, pagination_response::PaginationResponse};

#[derive(Clone, Debug, Default, PartialEq, Deserialize, Serialize)]
pub struct MetricListResponse {
    pub data: Vec<MetricSummary>,

    pub pagination_meta: PaginationResponse,
}

impl MetricListResponse {
    pub fn new(data: Vec<MetricSummary>, pagination_meta: PaginationResponse) -> Self {
        Self {
            data,
            pagination_meta,
        }
    }
}

File: typescript/src/models/metricListResponse.ts
// this file is @generated
import { type MetricSummary, MetricSummarySerializer } from "./metricSummary";
import {
  type PaginationResponse,
  PaginationResponseSerializer,
} from "./paginationResponse";

export interface MetricListResponse {
  data: MetricSummary[];

  paginationMeta: PaginationResponse;
}

export const MetricListResponseSerializer = {
  _fromJsonObject(object: any): MetricListResponse {
    return {
      data: object["data"].map((item: any) =>
        MetricSummarySerializer._fromJsonObject(item)
      ),
      paginationMeta: PaginationResponseSerializer._fromJsonObject(
        object["pagination_meta"]
      ),
    };
  },

  _toJsonObject(self: MetricListResponse): any {
    return {
      data: self.data.map((item: any) => MetricSummarySerializer._toJsonObject(item)),
      pagination_meta: PaginationResponseSerializer._toJsonObject(self.paginationMeta),
    };
  },
};
